src/Hosting/Systemd/test/UseSystemdTests.cs

```diff
@@ -22,5 +38,54 @@ public void DefaultsToOffOutsideOfService()
                 Assert.IsType<ConsoleLifetime>(lifetime);
             }
         }
+
+
+        private static bool CheckSystemdUnit(ITestOutputHelper output)
+        {
+            // No point in testing anything unless it's Unix
+            if (Environment.OSVersion.Platform != PlatformID.Unix)
+            {
+                output.WriteLine("Platform not unix");
+                return false;
+            }
+
+            // We've got invocation id, it's systemd >= 232 running a unit (either directly or through a child process)
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(INVOCATION_ID)))
```

Comment:
@Tratcher Looks like this check isn't functioning properly on ubuntu 18.04. From the test logs:
```
[m[31;1m    Microsoft.Extensions.Hosting.UseSystemdTests.DefaultsToOffOutsideOfService [FAIL]
[m[37m      Assert.IsType() Failure
[m[37m      Expected: Microsoft.Extensions.Hosting.Internal.ConsoleLifetime
[m[37m      Actual:   Microsoft.Extensions.Hosting.Systemd.SystemdLifetime
[m[30;1m      Stack Trace:
[m[37m        /_/src/Hosting/Systemd/test/UseSystemdTests.cs(38,0): at Microsoft.Extensions.Hosting.UseSystemdTests.DefaultsToOffOutsideOfService()
[m[30;1m      Output:
[m[37m        Invocation ID obtained: 2303132d69eb43498bc4ce6a0556d9ae
```

Looks like invocation_id is not empty even though we didn't launch via systemd. I'm not sure how updating to ubuntu18.04 can cause this nor did a quick search yield any insights. Any ideas on how to investigate further?